### DISCUSSION TITLE
Model Kotlin sealed class in OpenAPI

### DISCUSSION THREAD

**Discussion by hfhbd** [nomisRev/OpenAPI-kt#82] [Q&A]
Hey,
just curios, do you support the generation of OpenAPI files (like I do for basic classes)? 

f so, how do you model Kotlin sealed classes in OpenAPI? I didn't implement sealed classes yet, but using the same representation could be useful.

**Comment by nomisRev** [comment]
Yes, I support sealed classes. I support oneOf, anyOf, allOf, and open enumerations.
It's kind-of tricky to explain since that is probably the most complex part of the `typed` module that turns the `OpenAPI` typed value into a model that is sensible to generate the code.

[Here](https://github.com/nomisRev/OpenAPI-kt/blob/f681ed7fd32ad8ddd8e668d99c1980da394494e6/typed/src/commonMain/kotlin/io/github/nomisrev/openapi/OpenAPITransformer.kt#L148) is where I attempt to detect the models. Sorry, the code is a bit messy still.
